cloudant-client/src/main/java/com/cloudant/client/api/ClientBuilder.java

```diff
@@ -193,14 +193,21 @@ private ClientBuilder(URL url) {
                     .getUserInfo()
                     .indexOf(":") + 1);
         }
-
         try {
             //Remove user credentials from url
-            this.url = new URL(urlProtocol
+            String tmpUrl = urlProtocol
                     + "://"
                     + urlHost
                     + ":"
-                    + urlPort);
+                    + urlPort;
+            //add url path if it exist in source url
```

Comment:
I'd prefer to see all this path manipulation code kept together rather than being interspersed with the construction of the new `URL`. Then the new URL creation code doesn't have to change much from before, since we know the urlPath will either be `""` or cleaned up already and the code is a bit more clear with the responsibility of each paragraph to handle part of the URL before they are used to make the new one.

e.g.
```java
        // Check if a path exists and sanitize it by removing whitespace and any trailing /
        String urlPath = url.getPath().trim();
        urlPath = urlPath.endsWith("/") ? urlPath.substring(0, urlPath.length() - 1) : urlPath;

        try {
            // Reconstruct URL without user credentials
            this.url = new URL(urlProtocol
                    + "://"
                    + urlHost
                    + ":"
                    + urlPort
                    + urlPath);
        } catch (MalformedURLException e) {
            throw new RuntimeException(e);
        }
```